src/storage/src/store_impl.rs

```diff
@@ -542,12 +542,15 @@ impl StateStoreImpl {
         storage_metrics: Arc<MonitoredStorageMetrics>,
         compactor_metrics: Arc<CompactorMetrics>,
     ) -> StorageResult<Self> {
+        assert!(set_metrics_registry(GLOBAL_METRICS_REGISTRY.clone()));
```

Comment:
When will `set_metrics_registry` return false?

Reply:
When the metrics registry is already set. The metrics in foyer will be registered when opening. So `set_metrics_registry` is supposed to be called before it and be called no more than once.